nix/README.md

```diff
@@ -0,0 +1,102 @@
+
+## why Nix
+
+The Nix ecosystem provides tooling to enable declarative specification and replicable application of much (and growing) of my dotfiles in the Nix language.
+
+Pre-Nix these dotfiles applied themselves via shell scripts. I actually kind of enjoy writing shell scripts (shellcheck, shfmt, and a bash language server make a huge difference for me) but that's still a fragile system supported only by me.
+
+Instead of that fragile collection of imperative shell scripts Nix enables a flexible, composable configuration that glues together disparate bits of config with a more consistent structure.
+
+## how it works at a high-level
+
+In practice a configuration boils down to putting files in expected locations. For example, `zsh` will respect any configuration in a `$HOME/.zshrc` file (and others like `/etc/zshrc`, `$HOME/.zshenv`, etc.). These files may be read directly by the program or they may set features of an environment (like environment variables or shell aliases) that apply to the tool when run in that environment. Even the act of installing a package in the end just comes down to putting the corresponding binary in a folder on your `$PATH`.
+
+> There are, of course, exceptions to this general principle. macOS system settings for example are written via `defaults write` which updates a SQLite (TODO: double check this) database.
+
+The complexity of a dotfiles configuration comes primarily from the sheer variety of files/expected locations and the occasional need to interface with a third party tool to move a given file to its corresponding location. For example, to install a given package one needs to interface with a package manager (`apt`, `yum`, `brew`, etc.).
+
+Nix tooling abstracts this complexity away by providing a consistent interface to declare a wider range of config than any other tool I'm aware of. This varies from the least abstract symlinking of arbitrary files to setting high-level keys in a corresponding module which Nix tooling realizes by writing the corresponding files.
+
+In the high-level key case the Nix ecosystem has created a wrapper around a bit of config that makes declaring it as simple as setting the value of a specific key. For example, this [`system.defaults.NSGlobalDomain._HIHideMenuBar`](TODO: fill this) key controls whether or not the macOS menu bar is hidden. This [`users.users.nathan.packages`](TODO: fill this) key specifies the set of packages available to the `nathan` user. Nix tooling performs the actions required to apply this config: writing files, installing a package, flipping a macOS setting, etc.
+
+In other situations it may make more sense to declare a bit of config in its own file. To apply this config the file needs to be moved to the expected location for it. For example, [this key](TODO: fill this) causes Nix to apply the [`tmux.conf`](../tmux/tmux.conf) config in these dotfiles by moving it to `~/.tmux.conf`.
```

Comment:
"this key" means nothing to me because i have not yet been introduced to a Nix config.

Reply:
also, who is the intended audience here? if it's a dev who is interested in this stuff, i recommend assuming they know how "config at large" works, since so far most of the content here has been explaining how config works (flags, files, etc)

Reply:
that'd let you cut a lot of the preamble, and get straight to what i'm here for: what is Nix and why do I care?